File: Front.QuizApp/src/app/business/mappers/user.mapper.ts
import {Injectable} from '@angular/core';
import {UserDto} from '../dtos/user.dto';
import {User} from '../models/user.model';

@Injectable({providedIn: 'root'})
export class UserMapper {
    public mapUserFromApiToModel(apiUser: UserDto): User {
        return {
            id: apiUser.id,
            firstName: apiUser.firstName,
            lastName: apiUser.lastName,
            email: apiUser.email,
        };
    }
}

File: Back.QuizApp/Controllers/Mappers/UserMapper.cs
using Ynov.QuizApp.Business.Models;
using Ynov.QuizApp.Controllers.DTOs;

namespace Ynov.QuizApp.Controllers.Mappers;

public class UserMapper {
    public UserDTO ToDto(User user) {
        return new UserDTO() {
            Id = user.Id,
            FirstName = user.FirstName,
            LastName = user.LastName,
            Email = user.Email
        }; 
    }
}